src/mono/wasm/debugger/BrowserDebugProxy/EvaluateExpression.cs

```diff
@@ -222,59 +215,56 @@ private object ConvertJSToCSharpType(JToken variable)
                 JToken value = variable["value"];
                 string type = variable["type"].Value<string>();
                 string subType = variable["subtype"]?.Value<string>();
-
+                string objectId = variable["objectId"]?.Value<string>();
                 switch (type)
                 {
                     case "string":
-                        return value?.Value<string>();
+                        return $"\"{value?.Value<string>()}\"";
                     case "number":
                         return value?.Value<double>();
                     case "boolean":
-                        return value?.Value<bool>();
+                        return value?.Value<string>().ToLower();
                     case "object":
-                        return variable;
+                    {
+                        return "Newtonsoft.Json.Linq.JObject.FromObject(new {"
+                            + $"type = \"{type}\""
+                            + $", description = \"{variable["description"].Value<string>()}\""
+                            + $", className = \"{variable["className"].Value<string>()}\""
+                            + (subType != null ? $", subtype = \"{subType}\"" : "")
+                            + (objectId != null ? $", objectId = \"{objectId}\"" : "")
+                            + "})";
+                    }
                     case "void":
-                        return null;
+                        return "Newtonsoft.Json.Linq.JObject.FromObject(new {"
+                                + $"type = \"object\","
+                                + $"description = \"object\","
+                                + $"className = \"object\","
+                                + $"subtype = \"null\""
+                                + "})";
                 }
                 throw new Exception($"Evaluate of this datatype {type} not implemented yet");//, "Unsupported");
             }
 
             private string GetTypeFullName(JToken variable)
             {
                 string type = variable["type"].ToString();
-                string subType = variable["subtype"]?.Value<string>();
                 object value = ConvertJSToCSharpType(variable);
 
                 switch (type)
                 {
                     case "object":
-                        {
-                            if (subType == "null")
-                                return variable["className"].Value<string>();
-                            else
-                                return "object";
-                        }
+                        return "object";
                     case "void":
                         return "object";
+                    case "boolean":
```

Comment:
add this case to the tests too

Reply:
This line is testing it:                    ("this.CallMethodWithParmBool(this.t)", TString("TRUE")),

Reply:
Ah, and this was being handled by `default: return value.GetType().FullName;` earlier? Why the additional case then?

Reply:
Because I did this change:
![image](https://user-images.githubusercontent.com/4503299/141372101-39210957-0c30-4eaf-9139-855680717c20.png)

And this change was needed because otherwise when the variable was being declared and used it was creating like this: `var exampleVar4893742 = True;` and the first letter of the True/False with uppercase throws a compilation error.

Reply:
ok, that makes sense. IIUC, `ConvertJSToCSharpType` is being called once to get the value, and then `GetTypeFullName` also calls it. Instead, I think it would make sense for `ConvertJSToCSharpType` to return a `(string type, string value)`, since the type that we want to use in the final expression is tied closely to the "value" that we return.

And the method can be renamed to something like `ConvertJSToCSharpVariable`.